src/top/data/colored_cube_dataset.py

```diff
@@ -209,19 +213,32 @@ def __iter__(self):
             # TODO(ycho): Figure out a way to unify these formats.
             # see ai604-video-object-pose#10
             out = {
-                'image': img,
-                'object/orientation': orientation,
-                'object/translation': T,
-                'object/scale': th.full_like(T, 1.0),
-                'points': points_2d,
-                'num_instances': th.ones(
+                Schema.IMAGE: img,
+                # NOTE(ycho): For now, only have `1` object per image.
+                Schema.CLASS: th.zeros((self.opts.batch_size, 1), dtype=th.int32,
+                                       device=self.device),
+                Schema.ORIENTATION: orientation,
+                Schema.TRANSLATION: T,
+                Schema.SCALE: th.full_like(T, 1.0),
+                Schema.KEYPOINT_2D: points_2d,
+                Schema.INSTANCE_NUM: th.ones(
                     self.opts.batch_size,
                     device=self.device),
-                'camera/projection': projection,
+                Schema.PROJECTION: projection,
             }
-            if self.xfm is not None:
-                out = self.xfm(out)
-            yield out
+
+            # Unstack the batched render into a set of images, for compatibility with Objectron.
+            # NOTE(ycho): See if this results in a significant performance hit.
+            if self.opts.unstack:
+                for i in range(self.opts.batch_size):
+                    out_i = {k: v[i] for k, v in out.items()}
+                    if self.xfm is not None:
+                        out_i = self.xfm(out_i)
+                    yield out_i
+            else:
+                if self.xfm is not None:
+                    out = self.xfm(out)
+                yield out
```

Comment:
What exactly does `unstack` mean? Is it correct to understand that it is a process of yielding images one by one when batch size images are bundled and processed?

Reply:
Yep. The thing here is we're rendering a batch of images, yet `Objectron` yields images one by one, thus we just make the dataset also yield images one by one for compatibility.